base/utils/R/read.output.R

```diff
@@ -16,160 +16,241 @@
-##' @param pft.name character string, name of the pft to read pft-specific output. If NULL no pft-specific output will be read even the variable has pft as a dimension.
-##' @return If \code{dataframe=FALSE}, a vector of output variables. If \code{dataframe=TRUE}, a data.frame of output variables with POSIXct timestamps added.
+##' For more details, see the [MsTMIP
+##' variables](http://nacp.ornl.gov/MsTMIP_variables.shtml)
+##' documentation.
+##'
+##' @param runid the ID distinguishing the model run. Can be omitted
+##'   if `ncfiles` is set.
+##' @param outdir the directory that the model's output was sent to.
+##'   Can be omitted if `ncfiles` is set.
+##' @param variables Character vector of variables to be read from
+##'   model output. Default = `"GPP"`. If `NULL`, try to read all
+##'   variables in output file..
+##' @param dataframe Logical: if TRUE, will return output in a
+##'   `data.frame` format with a posix column. Useful for
+##'   [PEcAn.benchmark::align.data()] and plotting.
+##' @param pft.name character string, name of the plant functional
+##'   type (PFT) to read PFT-specific output. If `NULL` no
+##'   PFT-specific output will be read even the variable has PFT as a
+##'   dimension.
+##' @param ncfiles Custom character vector of full paths to NetCDF
+##'   files. If `NULL` (default), this list is constructed
+##'   automatically by looking for `YYYY.nc` files in
+##'   `file.path(outdir, runid)`.
+##' @param verbose Logical. If `TRUE`, print status as every year and
+##'   variable is read (default = `FALSE`).
+##' @param print_summary Logical. If `TRUE` (default), calculate and
+##'   print a summary of the means of each variable for each year.
+##' @param start.year,end.year first and last year of output to read.
+##'   Specify as a date-time (only the year portion is used) or as a
+##'   four-digit number or string. If `NA`, reads all years found in
+##'   `outdir`.
+##' @return If `dataframe = FALSE`, a vector of output variables. If
+##'   `dataframe = TRUE`, a `data.frame` of output variables with
+##'   POSIXct timestamps added (`posix` column).
 ##' @export
-##' @author Michael Dietze, David LeBauer
-read.output <- function(runid, outdir, start.year = NA, end.year = NA, variables = "GPP", dataframe= FALSE, pft.name = NULL) {
-  
-  ## vars in units s-1 to be converted to y-1 cflux = c('GPP', 'NPP', 'NEE',
-  ## 'TotalResp', 'AutoResp', 'HeteroResp', 'DOC_flux', 'Fire_flux') # kgC m-2 s-1
+##' @author Michael Dietze, David LeBauer, Alexey Shiklomanov
+read.output <- function(runid, outdir,
+                        start.year = NA,
+                        end.year = NA,
+                        variables = "GPP",
+                        dataframe = FALSE,
+                        pft.name = NULL,
+                        ncfiles = NULL,
+                        verbose = FALSE,
```

Comment:
In most of our other functions that touch CDFs and have a `verbose` argument, it gets passed along to NCDF4. Is there any potential for confusion from this one being different?

(I'm leaning toward "no", but not strongly)

Reply:
I didn't realize that, but that's good to know. I agree that our arguments should be consistent (if for no other reason than to reuse documentation via `@inheritParams`).

It may make sense to have `verbose` serve both purposes here. Chances are if you're already interested in looking at verbose NetCDF output, you won't mind also seeing which variable is being accessed. Alternatively, I could create an argument along the lines of `print_progress`.

Reply:
Either of those seems workable to me, whichever comes closer to producing the output you wanted. Will there be cases where you want variables reported but *don't* want the chatter from NCDF4?

Reply:
> Will there be cases where you want variables reported but don't want the chatter from NCDF4?

Maybe, but I'm gonna punt on that very specific case for now. The other comments have been addressed in 7436896537511dd1172559f0931618737d2d38bf.